website/blog/2021-10-29-7.16.0.md

```diff
@@ -0,0 +1,88 @@
+---
+layout: post
+title:  "7.16.0 Released: ESLint 8 and TypeScript 4.5"
+author: Babel Team
+authorURL: https://twitter.com/babeljs
+date:   2021-10-29 0:00:00
+categories: announcements
+share_text: "Babel 7.16.0 Released"
+---
+
+We just published Babel 7.16.0!
+
+This release enables [class static initialization blocks](https://github.com/tc39/proposal-class-static-block) by default. It includes support for a new variant of the [pipeline operator](https://github.com/tc39/proposal-pipeline-operator) proposal, as well as TypeScript 4.5 compatibility.
+
+Furthermore, `@babel/eslint-parser` now supports ESLint 8.
+
+You can read the whole changelog [on GitHub](https://github.com/babel/babel/releases/tag/v7.16.0).
+
+<!-- truncate -->
+
+If you or your company want to support Babel and the evolution of JavaScript, but aren't sure how, you can donate to us on our [Open Collective](https://github.com/babel/babel?sponsor=1) and, better yet, work with us on the implementation of [new ECMAScript proposals](https://github.com/babel/proposals) directly! As a volunteer-driven project, we rely on the community's support to fund our efforts in supporting the wide range of JavaScript users. Reach out at [team@babeljs.io](mailto:team@babeljs.io) if you'd like to discuss more!
+
+## Highlights
+
+### Class static blocks enabled by default ([#13713](https://github.com/babel/babel/pull/13713))
+
+```js
+class MyClass {
+  static {
+    doSomeInitialization(MyClass);
+    console.log("MyClass initialized!");
+  }
+}
+```
+
+Static blocks, that [reached Stage 4 in August](https://github.com/tc39/notes/blob/HEAD/meetings/2021-08/aug-31.md#class-static-initialization-blocks-for-stage-4), are now enabled by default in `@babel/parser` and `@babel/preset-env`.
+
+If you were using the `classStaticBlock` parser plugin, or `@babel/plugin-syntax-class-static-block`, you can safely remove them from your config.
+
+If you already use `@babel/preset-env`, you can now remove `@babel/plugin-proposal-class-static-block` from your config.
+
+### TypeScript 4.5 features ([#13802](https://github.com/babel/babel/pull/13802), [#13838](https://github.com/babel/babel/pull/13838))
+
+TypeScript 4.5 introduces a new syntax for marking imports as type-only: rather than marking the whole import statement, you can mark a single specifier:
+```ts
+// TypeScript 4.4
+import type { Foo } from "my-module";
+import { fooInstance } from "my-module";
+
+// TypeScript 4.5
+import { type Foo, fooInstance } from "my-module";
+```
+
+It also supports two new file extensions: `.mts` and `.cts`, that mirror `.mjs` and `.cjs`. When passing an `.mts` or `.cts` file to Babel with `@babel/preset-typescript` enabled, it uses the file extension to detect the desired source type (`"module"` or `"script"`).
+
+`.mts` and `.cts` file cannot contain JSX code by default, but they cannot contain TypeScript annotations that would be ambiguous with JSX (`<Type> cast` and `<T>() => {}`).
```

Comment:
Okay. Guess we should also update `allExtensions` in https://github.com/babel/website/pull/2585. I think users  recognize that `allExtensions` treats JS files as `TS` since generally TS is a superset of JS, but not sure if `mts` should be treated as `ts`, too.

Reply:
Docs updated :+1: https://github.com/babel/website/pull/2585

I think `mts` should be treated as `ts` because `allExtensions` is needed to disable extension detection, and it must also work when not providing a filename.